pkg/controller/storageversionmigrator/storageversionmigrator.go

```diff
@@ -364,6 +381,13 @@ func (svmc *SVMController) runMigration(ctx context.Context, logger klog.Logger,
 		}
 		logger.V(4).Info("Successfully migrated the resource", "namespace", accessor.GetNamespace(), "name", accessor.GetName(), "gvr", gvr.String())
 	}
+	// Finish migrating CRD if it exists
+	if crd != nil {
+		if err := svmc.finishCRDMigration(ctx, crd); err != nil {
```

Comment:
if there are two SVMs that identify the same group/resource connected to the same CRD, I think this logic can do the wrong thing... we could record the condition for one SVM at one generation, then mark the CRD as migrated and prune status.storedVersions when a different migration completed